### PR TITLE
Luke/React to v15.0.1

### PR DESCRIPTION
**Pull request by lukeschunk**
When I was trying to update Gunny to React `v15.0.1` I got an error that involved two different versions of React Installed - `v0.14.x` from the open source, and `v15.0.1` from gunny. 

This PR updates React in the Open Source. It also addresses all the errors in the demo app that were caused by the update. 
* React 15 doesn't like strings of numbers without px: ie `size='20'` is bad - we decided on `size={20}`
* The DatePickerFullScreen was mad that the value was null in some cases, I emulated the normal DatePicker and added getInitialState to the full screen version. This fixed the error. 

I noticed some other things that need to be cleaned up in the components - ie potential unused state in the datePicker, but that's for another PR. 

### Important:
Because this updates our OSS components to `react 15.0.1` i'm assuming this would be a major version bump? Any thoughts on this?

### CODE REVIEW

**Comment by jmophoto** [comment]
Thinking this might have the reverse effect for the duplicate React versions for the projects that aren't updated. I would say yes to bumping the major version.

File: src/components/DatePickerFullScreen.js [hunk 1]
```diff
@@ -40,6 +40,16 @@ const DatePickerFullScreen = React.createClass({
     };
   },
 
+  getInitialState () {
```
Comment (iheartrachie) [inline src/components/DatePickerFullScreen.js]: Should this be `getDefaultProps` instead?

Comment (jmophoto) [inline src/components/DatePickerFullScreen.js]: There are `defaultProps`, too. These are states that only the component handles.

**Comment by cerinman** [comment]
Needs master merged and conflicts resolved.  I agree that this should be a major version bump.

**Comment by lukeschunk** [comment]
This has been merged with master

**Comment by cerinman** [comment]
:shipit: Let make sure that we we publish this we get the change log updated right away about the React 15 upgrade and how it might affect users that use our components with an app that has an early version of React.  Also make sure we give everyone internally a heads up so that they know about needing to update React internally before using this as well.

**Pull request merged**